### PR TITLE
Личный проект: подробности

### PR DESCRIPTION
**Pull request by gromniki**
---
:mortar_board: [Личный проект: подробности](https://up.htmlacademy.ru/javascript/17/user/213549/tasks/7)

:boom: https://htmlacademy-javascript.github.io/213549-keksobooking-17/3/

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -1,15 +1,61 @@
 'use strict';
 
+var MAIN_PIN_WIDTH = 65;
+var MAIN_PIN_HEIGHT = 65;
+var centerMainPinX = MAIN_PIN_WIDTH / 2;
+var centerMainPinY = MAIN_PIN_HEIGHT / 2;
```
Comment (rkhaslarov) [inline js/main.js]: if we divide the constant and constant value it's a constant too =)

File: js/main.js [hunk 2]
```diff
@@ -1,15 +1,61 @@
...
+  return array;
+};
+
+var delDisabled = function (array) {
+  for (var i = 0; i < array.length; i++) {
+    array[i].removeAttribute('disabled', '');
+  }
+
+  return array;
+};
```
Comment (rkhaslarov) [inline js/main.js]: I think, these two function can be merged together

File: js/main.js [hunk 3]
```diff
@@ -1,15 +1,61 @@
...
+
+mainPin.addEventListener('click', function () {
+  map.classList.remove('map--faded');
+  adForm.classList.remove('ad-form--disabled');
+
+  delDisabled(mapFilters);
+  delDisabled(fieldsetsForm);
+
+  renderPins(generatePinsData());
+});
```
Comment (rkhaslarov) [inline js/main.js]: strange place for the global code, moreover it'd be better to place inside the functions

File: js/main.js [hunk 4]
```diff
@@ -1,15 +1,61 @@
...
+  return array;
+};
+
+setDisabled(mapFilters);
+setDisabled(fieldsetsForm);
+
+var topMainPin = parseInt(mainPin.style.top, 10) + centerMainPinY;
+var leftMainPin = parseInt(mainPin.style.left, 10) + centerMainPinX;
+
+fieldAddress.value = leftMainPin + ', ' + topMainPin;
```
Comment (rkhaslarov) [inline js/main.js]: place the code related to setting coords inside the function

File: js/main.js [hunk 5]
```diff
@@ -63,6 +90,17 @@ var renderPins = function (array) {
...
 
-renderPins(generatePinsData());
+mainPin.addEventListener('click', function () {
+  map.classList.remove('map--faded');
+  adForm.classList.remove('ad-form--disabled');
+
+  defineDisabled(mapFilters, true);
+  defineDisabled(fieldsetsForm, true);
+
+  renderPins(generatePinsData());
```
Comment (rkhaslarov) [inline js/main.js]: still can see the problem with pins, try to click twice to the main pin and you'll see 16 pins on the map

File: js/main.js [hunk 6]
```diff
@@ -28,15 +55,15 @@ var generatePinsData = function () {
...
       offer: {
-        type: getRandomElement(TYPES),
+        type: getRandomElement(TYPES)
       },
       location: {
-        x: getRandomNumber(0, 1200),
-        y: getRandomNumber(130, 630),
-      },
+        x: getRandomNumber(0, 1130),
+        y: getRandomNumber(130, 630)
```
Comment (rkhaslarov) [inline js/main.js]: Would be better to create some config object for all these magic values, 0, 1130, 130, 630

File: js/main.js [hunk 7]
```diff
@@ -63,6 +90,21 @@ var renderPins = function (array) {
...
+
+  renderPins(generatePinsData());
+
+  mainPin.removeEventListener('click', onMapPinMainClick);
+}
+
+mainPin.addEventListener('click', onMapPinMainClick);
+
+defineDisabled(mapFilters, false);
+defineDisabled(fieldsetsForm, false);
```
Comment (rkhaslarov) [inline js/main.js]: setDisabled will be nice for the name, define sounds strange

File: js/main.js [hunk 8]
```diff
@@ -1,19 +1,46 @@
...
+      array[i].removeAttribute('disabled', '');
+    } else {
+      array[i].setAttribute('disabled', '');
+    }
+  }
+
+  return array;
+};
+
+var setCoords = function () {
```
Comment (rkhaslarov) [inline js/main.js]: setAddressValue?

File: js/main.js (hunk 7 above)
Comment (gromniki) [reply js/main.js]: ok. Rightly

**Pull request merged**